sky/client/cli/command.py

```diff
@@ -4659,7 +4659,9 @@ def _generate_task_with_service(
                         'explicitly set the service port in the '
                         '`service.ports` section.')
         assert service_port is not None
-        task.service.set_ports(str(service_port))
+        service = task.service
+        service.set_ports(str(service_port))
+        task.set_service(service)
```

Comment:
Same as the `_file_mounts`, this will cause the `_service` to be set with the substitute version?